objects/src/block/note_tree.rs

```diff
@@ -0,0 +1,53 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, SimpleSmt},
+};
+
+use crate::{notes::NoteMetadata, BLOCK_OUTPUT_NOTES_TREE_DEPTH, MAX_NOTES_PER_BATCH};
+
+/// Wrapper over [SimpleSmt] for notes tree
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct BlockNoteTree(SimpleSmt<BLOCK_OUTPUT_NOTES_TREE_DEPTH>);
```

Comment:
Similar comment re adding more documentation.

Also, should we add `Serializable` and `Deserializable` implementations to this struct so that we can send it over the wire more easily in the node?